core/src/auth_tokens.rs

```diff
@@ -201,36 +202,15 @@ impl BearerToken {
     /// Validate that a bearer token value matches the format in
     /// https://datatracker.ietf.org/doc/html/rfc6750#section-2.1.
     fn validate(value: &str) -> Result<(), anyhow::Error> {
-        let mut iter = value.chars();
-        let mut any_non_equals = false;
-        // First loop: consume "normal" characters, stop when we see an equals sign for padding or
-        // reach the end of the input.
-        for c in &mut iter {
-            match c {
-                'a'..='z' | 'A'..='Z' | '0'..='9' | '-' | '.' | '_' | '~' | '+' | '/' => {
-                    any_non_equals = true;
-                }
-                '=' => {
-                    if !any_non_equals {
-                        return Err(anyhow::anyhow!("bearer token may not start with '='"));
-                    }
-                    break;
-                }
-                _ => return Err(anyhow::anyhow!("bearer token may not contain '{c}'")),
-            }
-        }
-        // Second loop: consume any further padding characters, if present.
-        for c in &mut iter {
-            match c {
-                '=' => {}
-                _ => {
-                    return Err(anyhow::anyhow!(
-                        "bearer token may only contain '=' at the end"
-                    ))
-                }
-            }
+        static REGEX: OnceLock<Regex> = OnceLock::new();
```

Comment:
Neat-o, `OnceLock` was stabilized in 1.70, which is now our MSRV. We can probably drop `lazy_static`, with a bit of effort.

Reply:
Already done in #1852 :grin:

Reply:
Aha, while I was on vacation :laughing: